apps/dashboard/pages/api/v1/github/verify.ts

```diff
@@ -0,0 +1,196 @@
+import { Buffer } from "node:buffer";
+import crypto from "node:crypto";
+import { db } from "@/lib/db";
+import { env } from "@/lib/env";
+
+import type { Readable } from "node:stream";
+import { clerkClient } from "@clerk/nextjs";
+import { sha256 } from "@unkey/hash";
+import { Resend } from "@unkey/resend";
+import type { NextApiRequest, NextApiResponse } from "next";
+export const config = {
+  api: {
+    bodyParser: false,
+  },
+  runtime: "nodejs",
+};
+const GITHUB_KEYS_URI = "https://api.github.com/meta/public_keys/secret_scanning";
+const { RESEND_API_KEY } = env(); // add RESEND_API_KEY
+
+const resend = new Resend({
+  apiKey: RESEND_API_KEY,
+});
+
+type Key = {
+  key_identifier: string;
+  key: string;
+  is_current: boolean;
+};
+// Needs to be tested when Github is live
+const verify_git_signature = async (payload: string, signature: string, keyID: string) => {
+  //Check payload
+  if (!payload || payload.length === 0) {
+    throw new Error("No payload found");
+  }
+  //Check signature
+  if (!signature || signature.length === 0 || typeof signature !== "string") {
+    throw new Error("No signature found");
+  }
+  //Check KeyID
+  if (!keyID || keyID.length === 0 || typeof keyID !== "string") {
+    throw new Error("No KeyID found");
+  }
+  //Get data from Github url
+  const gitHub = await fetch(GITHUB_KEYS_URI);
+  if (!gitHub.ok) {
+    throw new Error("No public keys found");
+  }
+  const gitBody = gitHub.ok ? await gitHub.json() : undefined;
+  //Get public keys from Github
+  const public_key =
+    gitBody.public_keys.find((k: Key) => {
+      if (k.key_identifier === keyID) {
+        return k;
+      }
+    }) ?? null;
+
+  if (!public_key) {
+    console.error("No public keys found");
+  }
+  // Verify signature
+  const verify = crypto.createVerify("SHA256").update(payload);
+  if (!verify.verify(public_key.key, Buffer.from(signature.toString(), "base64"))) {
+    return false;
+  }
+  return true;
+};
+
+export default async function handler(request: NextApiRequest, response: NextApiResponse) {
+  const signature = request.headers["github-public-key-signature"]?.toString();
+  const keyId = request.headers["github-public-key-identifier"]?.toString();
+  async function getRawBody(readable: Readable): Promise<Buffer> {
```

Comment:
Can we move this out of the API handler, seems like a random function to just stick in the middle.